File: Python3/Implement strStr().py
# 40ms, 100.0%
class Solution:
    def strStr(self, haystack, needle):
        if not needle: return 0
        trav, i, end1, end2 = 0, 0, len(haystack) - len(needle), len(needle) - 1
        while trav <= end1:
            while haystack[trav + i] == needle[i]:
                if i == end2: return trav
                else: i += 1
            else:
                trav, i = trav + 1, 0
        return -1
        

File: Go/Implement strStr().go
// 0ms, 100.00%; 2.3MB, 87.50%
func strStr(haystack string, needle string) int {
	end := len(haystack) - len(needle) + 1
	for i := 0; i < end; i++ {
		flag := true
		for j := 0; j < len(needle); j++ {
			if haystack[i+j] != needle[j] {
				flag = false
				break
			}
		}
		if flag {
			return i
		}
	}
	return -1
}
